app/services/service_offering/add_to_portfolio_item.rb

```diff
@@ -0,0 +1,56 @@
+module ServiceOffering
+  class AddToPortfolioItem
+    IGNORE_FIELDS = %w(id created_at updated_at portfolio_id)
+
+    attr_reader :params
+
+    def initialize(params)
+      @params = params
+    end
+
+    def process
+      # Get the api response object from Topology
+      TopologicalInventory.call do |api_instance|
+        @service_offering = api_instance.show_service_offering(@params[:service_offering_ref])
+      end
+
+      # Get the fields that we're going to pull over
+      fields = determine_valid_fields
+      params = create_param_map(fields)
+
+      populate_missing_fields(params)
+
+      PortfolioItem.create!(params)
```

Comment:
So, thinking about this some more.  What if lines 18 & 19 became:
```
create_param_map(determine_valid_fields)
```

Furthermore combine 21 & 23:
```
Portfolio.create!(populate_missing_fields)
```

Since `@params` is an instance variable it can be populated with the correct key/value pairs without the need to pass around a local.

Reply:
refactored to make it easier to read, like so.